pkgs/build-support/fake-uname/default.nix

```diff
@@ -0,0 +1,34 @@
+# expr and script based on our lsb_release
+{ stdenv
+, lib
+, substituteAll
+, coreutils
+, getopt
+, modDirVersion ? ""
+}:
+
+substituteAll {
+  name = "uname";
+
+  src = ./fake-uname.sh;
+
+  dir = "bin";
+  isExecutable = true;
+
+  inherit coreutils getopt;
+
+  uSystem = if stdenv.buildPlatform.uname.system != null then stdenv.buildPlatform.uname.system else "unknown";
+  inherit (stdenv.buildPlatform.uname) system processor;
+
+  # in os-specific/linux module packages
+  # --replace '$(shell uname -r)' "${kernel.modDirVersion}" \
+  # is a common thing to do.
+  modDirVersion = if modDirVersion != "" then modDirVersion else "unknown";
+
+  meta = with lib; {
+    description = "Print certain system information (hardcoded with NixOS values)";
+    license = [ licenses.mit ];
```

Comment:
```suggestion
        Description = ''
      This package provides a replacement for `uname` whose output depends only
      on `stdenv.buildPlatform`.  It is meant to be used from within derivations.

      Many packages' build processes run `uname` at compile time and embed its
      output into the result of the build.  Since `uname` calls into the kernel,
      and the Nix sandbox currently does not intercept these calls, builds made
      on different kernels will produce different results.
    '';
    license = [ licenses.mit ];
```